src/printer_tree.c

```diff
@@ -0,0 +1,3107 @@
+ */
+struct trt_pck_print {
+    const struct trt_tree_ctx *tree_ctx;    /**< Context of libyang tree. */
+    struct trt_fp_print fps;                /**< Print function. */
+};
+
+/******************************************************************************
+ * Indent
+ *****************************************************************************/
+
+/**
+ * @brief Constants which are defined in the RFC or are observable from the pyang tool.
+ */
+typedef enum {
+    TRD_INDENT_EMPTY = 0,               /**< If the node is a case node, there is no space before the \<name\>. */
+    TRD_INDENT_LONG_LINE_BREAK = 2,     /**< The new line should be indented so that it starts below \<name\> with a whitespace offset of at least two characters. */
+    TRD_INDENT_LINE_BEGIN = 2,          /**< Indent below the keyword (module, augment ...).  */
+    TRD_INDENT_BTW_SIBLINGS = 2,        /**< Indent between | and | characters. */
+    TRD_INDENT_BEFORE_KEYS = 1,         /**< "..."___\<keys\>. */
+    TRD_INDENT_BEFORE_TYPE = 4,         /**< "..."___\<type\>, but if mark is set then indent == 3. */
+    TRD_INDENT_BEFORE_IFFEATURES = 1    /**< "..."___\<iffeatures\>. */
+} trt_cnf_indent;
+
+/**
+ * @brief Type of indent in node.
+ */
+typedef enum {
+    TRD_INDENT_IN_NODE_NORMAL = 0,  /**< Node fits on one line. */
+    TRD_INDENT_IN_NODE_DIVIDED,     /**< The node must be split into multiple rows. */
+    TRD_INDENT_IN_NODE_FAILED       /**< Cannot be crammed into one line. The condition for the maximum line length is violated. */
+} trt_indent_in_node_type;
+
+/** Constant to indicate the need to break a line. */
+#define TRD_LINEBREAK -1
+
+/**
+ * @brief Records the alignment between the individual elements of the node.
+ *
+ * See trp_indent_in_node_are_eq, trp_indent_in_node_place_break.
+ */
+struct trt_indent_in_node {
+    trt_indent_in_node_type type;   /**< Type of indent in node. */
+    int16_t btw_name_opts;          /**< Indent between node name and opts. */
+    int16_t btw_opts_type;          /**< Indent between opts and type. */
+    int16_t btw_type_iffeatures;    /**< Indent between type and features. Ignored if \<type\> missing. */
+};
+
+/**
+ * @brief Type of wrappers to be printed.
+ */
+typedef enum {
+    TRD_WRAPPER_TOP = 0,    /**< Related to the module. */
+    TRD_WRAPPER_BODY        /**< Related to e.g. Augmentations or Groupings */
+} trd_wrapper_type;
+
+/**
+ * @brief For resolving sibling symbol ('|') placement.
+ *
+ * Bit indicates where the sibling symbol must be printed.
+ * This place is in multiples of TRD_INDENT_BTW_SIBLINGS.
+ *
+ * See: trp_init_wrapper_top, trp_init_wrapper_body, trp_wrapper_set_mark,
+ * trp_wrapper_set_shift, trp_wrapper_if_last_sibling, trp_wrapper_eq,
+ * trp_print_wrapper
+ */
+struct trt_wrapper {
+    trd_wrapper_type type;  /**< Location of the wrapper. */
+    uint64_t bit_marks1;    /**< The set bits indicate where the '|' character is to be printed. */
```

Comment:
Limits the printed schema tree depth to 64 bits. It is probably okay (not sure if even the current *libyang* did not have this limitation) but perhaps at least mention it somewhere. Also, please put asserts or internal errors on places where this could overflow (in case the depth is more than 64).